### PR TITLE
[Android] Compile apps for android

### PR DESCRIPTION
**Pull request by talregev**
Compile apps for android

### CODE REVIEW

**Comment by talregev** [comment]
@LocutusOfBorg This PR will break your patches.

File: samples/vboxmonitor/Makefile [hunk 1]
```diff
@@ -2,14 +2,31 @@
 
 # Change the following to match your installation
 BOINC_DIR = ../..
-BOINC_API_DIR = $(BOINC_DIR)/api
-BOINC_LIB_DIR = $(BOINC_DIR)/lib
-
-CXXFLAGS = -g \
+BOINC_SOURCE_API_DIR = $(BOINC_DIR)/api
```
Comment (AenBleidd) [inline samples/vboxmonitor/Makefile:5]: You probably added it here for consistency but this is not supposed to be build for Android.
I'd probably removed redundant code for Android

File: samples/vboxwrapper/Makefile [hunk 2]
```diff
@@ -2,14 +2,31 @@
...
 # Change the following to match your installation
 BOINC_DIR = ../..
-BOINC_API_DIR = $(BOINC_DIR)/api
-BOINC_LIB_DIR = $(BOINC_DIR)/lib
+BOINC_SOURCE_API_DIR = $(BOINC_DIR)/api
+BOINC_SOURCE_LIB_DIR = $(BOINC_DIR)/lib
+BOINC_SOURCE_ZIP_DIR = $(BOINC_DIR)/zip
 
-CXXFLAGS = -g \
+ifndef ANDROID
```
Comment (AenBleidd) [inline samples/vboxwrapper/Makefile:9]: You probably added it here for consistency but this is not supposed to be build for Android.
I'd probably removed redundant code for Android

File: android/clear_environment.sh [hunk 3]
```diff
@@ -3,14 +3,20 @@ set -e
...
-    echo Please run it from boinc/android
+    echo Please run it from boinc
+    echo Type: \'android/clear_environment.sh\'
     exit 1
 fi
 
-rm -rf ../3rdParty/buildCache/
-rm -rf ../Makefile
-rm -rf ../m4/Makefile
+if [ "hard" = "$1" ]; then
```
Comment (AenBleidd) [inline android/clear_environment.sh:13]: Instead of `hard` I'd suggest to use `full`

**Review by AenBleidd — CHANGES_REQUESTED** [review]
Looks like `samples\example_app\Makefile_android` is not needed anymore and should be rmoved

File: android/build_example_arm64.sh [hunk 4]
```diff
@@ -20,21 +21,35 @@ export BOINC_API_DIR="$BOINC/api"
...
+        ./configure --host=aarch64-linux --with-boinc-platform="aarch64-android-linux-gnu" --with-boinc-alt-platform="arm-android-linux-gnu" --prefix="$TCINCLUDES" --libdir="$TCINCLUDES/lib" --with-ssl="$TCINCLUDES" --enable-apps --disable-server --disable-manager --disable-client --disable-libraries --disable-shared --enable-static --enable-boinczip
+    fi
     if [ "$VERBOSE" = "no" ]; then
-        make -C $BOINC/samples/example_app/ -f Makefile_android --silent
+        make --silent
     else
-        make -C $BOINC/samples/example_app/ -f Makefile_android SHELL="/bin/bash -x"
-    fi
+        make SHELL="/bin/bash -x"
+    fi
```
[the remark this answers is not in the document]
Comment (talregev) [reply android/build_example_arm64.sh]: Fix

File: configure.ac [hunk 5]
```diff
@@ -1067,6 +1067,8 @@ dnl or OS2
 AM_CONDITIONAL(OS_OS2, [echo $host_os | grep '^os2' > /dev/null])
 AM_CONDITIONAL(OS_ARM_LINUX, [echo $host_alias | grep '^arm-linux' > /dev/null])
 AM_CONDITIONAL(OS_ARMV6_LINUX, [echo $host_alias | grep '^armv6-linux' > /dev/null])
+AM_CONDITIONAL(ANDROID, [test x"${ANDROID}" = xyes])
+
```
[the remark this answers is not in the document]
Comment (talregev) [reply configure.ac:1071]: Fix

File: samples/vboxmonitor/Makefile (hunk 1 above)
Comment (talregev) [reply samples/vboxmonitor/Makefile:5]: I keep

File: samples/vboxwrapper/Makefile (hunk 2 above)
Comment (talregev) [reply samples/vboxwrapper/Makefile:9]: I keep

File: android/clear_environment.sh (hunk 3 above)
Comment (talregev) [reply android/clear_environment.sh:13]: I changed

File: samples/example_app/Makefile [hunk 6]
```diff
@@ -86,3 +105,4 @@ slide_show: slide_show.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR
 	$(BOINC_ZIP_DIR)/libboinc_zip.a \
     $(LIBGLUT) $(LIBGLU) $(LIBJPEG) \
     -lGL -lX11 -lXmu -lm
+
```
[the remark this answers is not in the document]
Comment (talregev) [reply samples/example_app/Makefile]: I removed

File: android/clear_environment.sh [hunk 7]
```diff
@@ -3,14 +3,20 @@ set -e
...
+    echo Please run it from boinc
+    echo Type: \'android/clear_environment.sh\'
     exit 1
 fi
 
-rm -rf ../3rdParty/buildCache/
-rm -rf ../Makefile
-rm -rf ../m4/Makefile
+if [ "full" = "$1" ]; then
+    echo "full clean"
```
Comment (AenBleidd) [inline android/clear_environment.sh:14]: Comment is not fixed

**Review by AenBleidd — CHANGES_REQUESTED** [review]

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**